tools/benchmark_tool/openvino/tools/benchmark/benchmark.py

```diff
@@ -116,23 +94,26 @@ def infer(self, exe_network, batch_size, latency_percentile, progress_bar=None):
         while (self.niter and iteration < self.niter) or \
               (self.duration_seconds and exec_time < self.duration_seconds) or \
               (self.api_type == 'async' and iteration % self.nireq):
+            if not self.legacy_mode:
+                processed_frames += data_queue.get_next_batch_size()
             if self.api_type == 'sync':
-                infer_requests[0].infer()
-                times.append(infer_requests[0].latency)
+                if not self.legacy_mode:
+                    requests[0].set_tensors(data_queue.get_next_input())
+                requests[0].infer()
+                times.append(requests[0].latency)
             else:
-                infer_request_id = exe_network.get_idle_request_id()
-                if infer_request_id < 0:
-                    status = exe_network.wait(num_requests=1)
-                    if status != StatusCode.OK:
-                        raise Exception("Wait for idle request failed!")
-                    infer_request_id = exe_network.get_idle_request_id()
-                    if infer_request_id < 0:
-                        raise Exception("Invalid request id!")
-                if infer_request_id in in_fly:
-                    times.append(infer_requests[infer_request_id].latency)
+                idle_id = requests.get_idle_request_id()
+                if idle_id in in_fly:
+                    times.append(requests[idle_id].latency)
+                    self.latency_groups[requests.userdata[idle_id]].times.append(requests[idle_id].latency)
+                else:
+                    in_fly.add(idle_id)
+                group_id = data_queue.current_group_id
+                if self.legacy_mode:
+                    processed_frames += data_queue.batch_sizes[idle_id % data_queue.size]
```

Comment:
Actually, for legacy mode, all batch sizes should be equal to each other. So we can check it before and then use the old formula iteration * batch_size and avoid updating it in the loop

Reply:
done